sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -220,7 +222,17 @@ private[hive] class HiveClientImpl(
     hiveConf
   }
 
-  private val userName = conf.getUser
+  private val userName = try {
+    val doAs = sys.env.get("HADOOP_USER_NAME").orNull
+    val ugi = if (doAs != null && doAs.length() > 0) {
+      UserGroupInformation.createProxyUser(doAs, UserGroupInformation.getLoginUser())
+    } else {
+      UserGroupInformation.getCurrentUser
+    }
+    ugi.getShortUserName
+  } catch {
+    case _: LoginException => throw new LoginException("Can not get login user.")
```

Comment:
What's the purpose of this? Just let the original exception propagate.

Reply:
According to the previous comment, I guess the origin code was `IOException` wrapped `LoginException`:
https://github.com/apache/spark/pull/23952/files#r268466632

so if we want to follow the origin code, use IOException, otherwise, just remove try/catch.

Reply:
@vanzin @HeartSaVioR 
In hive source code, it throw two exception: `LoginException`, `IOException`. I wonder if `Exception` is ok here.

Reply:
I guess things will fail with logging exception in caller site, so I guess the recommendation is just removing try/catch altogether.

Hive source code has to do something (like wrapping as you've seen) because both `LoginException` and `IOException` are checked exception - given you're coding in Scala you can forget about that.

Reply:
> I guess things will fail with logging exception in caller site, so I guess the recommendation is just removing try/catch altogether.
> 
> Hive source code has to do something (like wrapping as you've seen) because both `LoginException` and `IOException` are checked exceptions - given you're coding in Scala you can forget about that.

Yes, you are right, it will fail with logging exception. Changed it.